pkgs/development/libraries/boost/generic.nix

```diff
@@ -30,6 +31,9 @@ assert enableShared || enableStatic;
 # Python isn't supported when cross-compiling
 assert enablePython -> stdenv.hostPlatform == stdenv.buildPlatform;
 assert enableNumpy -> enablePython;
+# Boost < 1.67 cannot be built with python3
+# https://github.com/boostorg/python/commit/660487c43fde76f3e64f1cb2e644500da92fe582
+assert enablePython && (python == python3) -> (lib.versionAtLeast version "1.68.0");
```

Comment:
I think that would break everything that uses python 3.8. Maybe python.version is useful for comparing?